cyclic.go

```diff
@@ -13,6 +17,13 @@ func (e *Executor) HasCyclicDep() bool {
 		defer delete(visits, name)
 
 		for _, d := range t.Deps {
+			if strings.HasPrefix(d, "^") {
+				call, err := ParseCall(d)
+				if err != nil {
+					return true
+				}
+				d = call.Name
```

Comment:
I would suggest we do this even earlier, when we parse the YAML/JSON/TOML. This would allow us to let YAML/TOML/JSON handle quotes for parameter values, and thus confuse the user less. This would also allow us to deprecate the "^" syntax  in favour of a more explicit one. For instance we could say we create a new type "instruction":

```go
 type Task struct {		
 	Cmds      []instruction
 	Deps      []instruction
        ...
}

type instruction struct {
        IsShellCmd bool
        Task string  // holds name of task or full shell command.
        Vars map[string]string
}

func (instr *instruction) UnmarshalJSON(data []byte) error {
        m := make(map[string]string)
        err := json.Unmarshal(data, m)
        if err =! nil {
                instr.IsShellCmd = true
                return json.Unmarshal(data, &instr.Task)
        }
        instr.IsShellCmd = false
        instr.Task = m["task"]   // the task name, including an empty value (""), will be evaluated later.
        delete(m["task"])
        instr.Vars = m
        }
}

// Equivalent for YAML/TOML if needed... 
```
 I don't know if YAML/TOML will "fall-back" to convert data to JSON and parse it to UnamrshalJSON if an UnmarshalJSON function is found? that would be crazy performance wise, but functional wise it might make sense if they did?